src/mem/ruby/system/GPUCoalescer.cc

```diff
@@ -599,6 +599,9 @@ GPUCoalescer::hitCallback(CoalescedRequest* crequest,
                 // data response is not needed.
                 case RubyRequestType_ATOMIC_NO_RETURN:
                     assert(pkt->isAtomicOp());
+                    data.popAtomicLogEntryFront();
```

Comment:
We need to explicitly free the memory here as it does in the ATOMIC_RETURN case.  Something like this would prevent a memory leak:

`log = data.popAtomicLogEntryFront();
delete [] log;`